### PR TITLE
feat: add Object Field Edit page sections

### LINKED ISSUE #2160: Object Field Edit - Add page sections
Warning: this issue is part of a series of issues aiming to create the "Object Field Edition flow". This is issue number 2.

## Context

See #1769

For this issue:
- In the "Object Field Edit" page, reuse the `SettingsObjectFieldFormSection` component to add a "Name and description" section with a title, a description, and a form.
- Reuse the `SettingsObjectFieldTypeSelectSection` component to add a "Type and values" section, but disable the select.
- Add a "Danger zone" section with a title, a description and a "Disable" button. See if the "Danger zone" section from #1911 can be reused.

## Figma

Image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11383%3A58516&mode=design&t=EynOEhEsP06qxkYt-1

## Behavior

**GIVEN** I am on an Object Field Edit page
**THEN** I see a "Name and description" section with a title, a description, and a form
**AND** I see a "Type and values" section
**AND** I see a "Danger zone" section

### LINKED ISSUE #2163: Object Field Edit - Disable field
Warning: this issue is part of a series of issues aiming to create the "Object Field Edition flow". This is issue number 4.

## Context

See #1769 

For this issue: in the "Object Field Edit" page, clicking on the "Disable" button disables the field (use the `useUpdateOneMetadataField` hook) and navigates back to the "Object Detail" page.

## Figma

Image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11383%3A58516&mode=design&t=EynOEhEsP06qxkYt-1

## Behavior

**GIVEN** I am on the "Object Field Edit" page
**WHEN** I click on the "Disable" button
**THEN** the field is disabled
**AND** I navigate to the "Object Detail" page

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #2160, Closes #2163

### CODE REVIEW

File: front/src/modules/metadata/hooks/useObjectMetadata.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { useFindManyMetadataObjects } from './useFindManyMetadataObjects';
 import { useUpdateOneMetadataObject } from './useUpdateOneMetadataObject';
 
 export const useObjectMetadata = () => {
```
Comment (lucasbordeau) [inline front/src/modules/metadata/hooks/useObjectMetadata.ts:9]: We should rename this useMetadataObject to be consistent with the existing naming, but in another PR.

File: front/src/pages/settings/data-model/SettingsObjectEdit.tsx [hunk 2]
```diff
@@ -80,61 +85,52 @@ export const SettingsObjectEdit = () => {
...
+        />
+        <Section>
+          <H2Title title="Danger zone" description="Disable object" />
+          <Button
+            Icon={IconArchive}
+            title="Disable"
+            size="small"
+            onClick={handleDisable}
+          />
+        </Section>
```
Comment (martmull) [inline front/src/pages/settings/data-model/SettingsObjectEdit.tsx:133]: It seems that the update does not work in the local app I launched with your branch, can you double check ? (The issue is when i try to update the description (for ex), my changes are not written in the input)

**Review by lucasbordeau — APPROVED** [review]

File: front/src/pages/settings/data-model/SettingsObjectDetail.tsx [hunk 3]
```diff
@@ -33,49 +34,51 @@ export const SettingsObjectDetail = () => {
...
-            onEdit={() => navigate('./edit')}
-          />
-        )}
+        <SettingsAboutSection
+          iconKey={activeObject.icon ?? undefined}
+          name={activeObject.labelPlural || ''}
+          isCustom={activeObject.isCustom}
+          onDisable={handleDisable}
+          onEdit={() => navigate('./edit')}
+        />
```
Comment (martmull) [inline front/src/pages/settings/data-model/SettingsObjectDetail.tsx:77]: Think we should not display 'edit' for non editable Standard Objects
image

File: front/src/pages/settings/data-model/SettingsObjectFieldEdit.tsx [hunk 4]
```diff
@@ -16,15 +49,33 @@ export const SettingsObjectFieldEdit = () => {
...
+              { children: activeField.label },
             ]}
           />
         </SettingsHeaderContainer>
+        <SettingsObjectFieldFormSection
+          disabled={!activeField.isCustom}
+          name={activeField.label}
+          description={activeField.description ?? undefined}
+          iconKey={activeField.icon ?? undefined}
+          onChange={() => undefined}
```
Comment (martmull) [inline front/src/pages/settings/data-model/SettingsObjectFieldEdit.tsx:64]: Is it normal that the input changes are not handled? Maybe you want to do that in another PR

Comment (thaisguigon) [reply front/src/pages/settings/data-model/SettingsObjectFieldEdit.tsx:64]: Yes this is addressed in this PR: https://github.com/twentyhq/twenty/pull/2245

File: front/src/pages/settings/data-model/SettingsObjectDetail.tsx (hunk 3 above)
Comment (thaisguigon) [reply front/src/pages/settings/data-model/SettingsObjectDetail.tsx:77]: According to the original issue the Standard objects also have the "Edit" option: https://github.com/twentyhq/twenty/issues/1765 (see "Edit an object" section).
I'd suggest you raise this concern with @Bonapara :)

File: front/src/modules/metadata/hooks/useObjectMetadata.ts (hunk 1 above)
Comment (thaisguigon) [reply front/src/modules/metadata/hooks/useObjectMetadata.ts:9]: I agree we could find a better name, however the `useMetadataObject` hook already exists in this directory.
